src/libs/ck-libs/ampi/ampi.C

```diff
@@ -9364,9 +9364,7 @@ AMPI_API_IMPL(int, MPI_Comm_free, MPI_Comm *comm)
     if (*comm != MPI_COMM_WORLD && *comm != MPI_COMM_SELF) {
       ampi* ptr = getAmpiInstance(*comm);
       ptr->barrier();
-      if (ptr->getRank() == 0) {
-        CProxy_CkArray(ptr->ckGetArrayID()).ckDestroy();
-      }
+      CProxy_ampi(ptr->ckGetArrayID())[ptr->getRank()].ckDestroy();
```

Comment:
I think it's nicer / more idiomatic to make this 'ptr->thisProxy[ptr->thisIndex].ckDestroy();'